src/relations/provider.py

```diff
@@ -196,14 +206,46 @@ def _create_ranger_client(self):
         ranger = ranger_client.RangerClient(ranger_url, ranger_auth)
         return ranger
 
-    def _delete_ranger_service(self, service_id):
+    def _delete_ranger_service(self, service_id, relation_id):
         """Delete service in Ranger.
 
         Args:
             service_id: the ID of the service to delete
+            relation_id: the id of the relation
         """
         ranger = self._create_ranger_client()
         retrieved_service = ranger.get_service_by_id(service_id)
 
-        if retrieved_service is not None:
-            ranger.delete_service_by_id(service_id)
+        if retrieved_service is None:
+            return
+
+        if self._has_custom_policies(
```

Comment:
question (blocking): Why do we care about preserving custom policies?

Reply:
What happens if you cannot access the UI and you need to nuke the relation? Does `--force` work?

Reply:
This change still allows you to remove the relation, and trino will no longer use ranger policies but default to it's own. The trino service will just be left in tact in Ranger for you to reconnect to if you add the relation without re-creating all the policies. Worst case scenario, if you can't access the UI, you just have a unused ranger services archive in Ranger which is providing no functionality.